File: crud/src/main/java/fr/cicco/crud/entity/Commande.java
package fr.cicco.crud.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Builder
public class Commande   {
    @Id()
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_livreur")
    private Utilisateur livreur;

    @OneToMany(mappedBy = "commande")
    private List<CommandePizza> commandePizzaList;
    @ManyToOne
    @JoinColumn(name = "client_id")
    private Utilisateur client;
    private String status;
    private LocalDateTime dateCommande;
    @ManyToOne
    @JoinColumn(name = "magasin_id")
    private Magasin magasin;
    @OneToMany(mappedBy = "commande")
    private List<CommandeMenu> commandeMenuList;
    @ManyToOne
    @JoinColumn(name="adresse_id")
    private Adresse adresse;

}

File: crud/springCLI/python_entities_template/commande.py

from ..datas.autocompletion import *

name = 'Commande'

entity = {
    
    'Commande' : {
            
        'id':{
            Type: Long,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'livreur':{
            Type: "Utilisateur",
            options:{
                'dto':[
                    {'exists':True, 'fields':["email"]}
                ]
        
            }
        },

        'commandePizzaList':{
            Type: "List<CommandePizza>",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'client':{
            Type: "Utilisateur",
            options:{
                'dto':[
                    {'exists':True, 'fields':["email"]}
                ]
        
            }
        },

        'status':{
            Type: "String",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'dateCommande':{
            Type: LocalDateTime,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'magasin':{
            Type: "Magasin",
            options:{
                'dto':[
                    {'exists':True, 'fields':["nom"]}
                ]
        
            }
        },

        'commandeMenuList':{
            Type: "List<CommandeMenu>",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'adresse':{
            Type: "Adresse",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        }

}


